src/Symfony/Component/Validator/Constraints/EnumValidator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * EnumValidator validates that the value is one of the expected values.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class EnumValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Enum) {
+            throw new UnexpectedTypeException($constraint, Enum::class);
+        }
+
+        if (!enum_exists($constraint->type) || !method_exists($constraint->type, 'from')) {
+            throw new ConstraintDefinitionException('The "type" must be an \BackedEnum.');
+        }
+
+        if (null === $value) {
+            return;
+        }
+
+        try {
+            $value = $constraint->type::from($value);
+        } catch (\ValueError) {
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setParameter('{{ choices }}', $this->formatValues($constraint->type::cases()))
```

Comment:
If `only` or `except` are set, this error message would enumerate the wrong correct values.